skimage/exposure/_adapthist.py

```diff
@@ -164,29 +164,29 @@ def _clahe(image, ntiles_x, ntiles_y, clip_limit, nbins=128):
 
     # Interpolate greylevel mappings to get CLAHE image
     ystart = 0
-    for y in range(ntiles_y + 1):
+    for y in range(ny + 1):
         xstart = 0
         if y == 0:  # special case: top row
             ystep = height / 2.0
             yU = 0
             yB = 0
-        elif y == ntiles_y:  # special case: bottom row
+        elif y == ny:  # special case: bottom row
             ystep = height / 2.0
-            yU = ntiles_y - 1
+            yU = ny - 1
             yB = yU
         else:  # default values
             ystep = height
             yU = y - 1
             yB = yB + 1
 
-        for x in range(ntiles_x + 1):
+        for x in range(nx + 1):
             if x == 0:  # special case: left column
                 xstep = width / 2.0
                 xL = 0
                 xR = 0
-            elif x == ntiles_x:  # special case: right column
+            elif x == nx:  # special case: right column
                 xstep = width / 2.0
-                xL = ntiles_x - 1
+                xL = nx - 1
                 xR = xL
```

Comment:
This whole section is pretty messy... Would you like to take this opportunity to:

- change this to be consistent with our current dimension-naming guidelines: (r, c) instead of (y, x)
- remove a level of nesting by using `itertools.product`:
   `for y in range(ny + 1): for x in range(nx + 1):`
   becomes
   `for r, c in itertools.product(range(nr + 1), range(nc + 1)):`
- fix a bit of the conditional logic. For example, `yU = y - 1` in two cases (bottom row and default), so do that outside the conditions and update only when `y == 0`. I might even suggest to just write a helper function, `_update_clahe_local_params`: `rstep, rU, rB = _update_clahe_local_params(r, nr, height, rB)`, since you are doing the exact same thing on both axes. This opens the door to an n-dimensional implementation in the near future. =)